editor/components/block-list/block.js

```diff
@@ -307,7 +323,10 @@ export class BlockListBlock extends Component {
 			}
 		} else {
 			this.props.onSelectionStart( this.props.uid );
-			this.props.onSelect();
+
+			if ( ! this.props.isSelected ) {
```

Comment:
Minor: instead of doing the checks every time we try to use onSelect, is doing the check on the onSelect creation in mapStateToProps using ownProps an option?
This has the advantage of making sure that future onSelect additions the check is not forgotten.

Reply:
>is doing the check on the onSelect creation in mapStateToProps using ownProps an option?

Not here, since `isSelected` doesn't exist in `ownProps`, it is provided through `mapStateToProps`.

It is technically possible using `mergeProps`, the third argument of `connect`:

https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options

...but it's a bit more (error-prone) work to override the default here, and I think it's sensible enough that the component itself should check `isSelected` prior to calling `onSelect`.